Source/JavaScriptCore/runtime/ISO8601.cpp

```diff
@@ -530,10 +530,38 @@ std::optional<int64_t> parseUTCOffsetInMinutes(StringView string)
 template<typename CharacterType>
 static bool canBeCalendar(const StringParsingBuffer<CharacterType>& buffer)
 {
-    // https://tc39.es/proposal-temporal/#prod-Calendar
-    // Calendar :
-    //     [u-ca= CalendarName]
-    return buffer.lengthRemaining() >= 6 && buffer[0] == '[' && buffer[1] == 'u' && buffer[2] == '-' && buffer[3] == 'c' && buffer[4] == 'a' && buffer[5] == '=';
+    // https://tc39.es/proposal-temporal/#sec-temporal-parseisodatetime
+    // Step 4(a)(ii)(2)(a):
+    //  Let key be the source text matched by the AnnotationKey Parse Node contained within annotation
+    // This just checks for '[', followed by an optional '!' (critical flag),
+    // followed by a valid key, followed by an '='.
+
+    int32_t len = buffer.lengthRemaining();
+    // Check for [! or [, followed by any number of 'a'-'z' or '-', followed by '='
+    int32_t i = 0;
+    if (i < len) {
+        if (buffer[i] != '[')
+            return false;
+        i++;
+    }
+    if (i < len) {
+        if (buffer[i] == '!')
+            i++;
+    }
+    // '_' allowed as first char
+    if (i < len) {
+        if (buffer[i] == '_')
+            i++;
+    }
```

Comment:
These three characters are necessary, then, we should first check `buffer.lengthRemaining` is more than 3 and check each character with index.

Reply:
The first one is necessary, but the second two are optional.